subsys/bluetooth/mesh/gen_ponoff_srv.c

```diff
@@ -99,13 +132,7 @@ static void set_on_power_up(struct bt_mesh_ponoff_srv *srv,
 		srv->update(srv, ctx, old, new);
 	}
 
-	struct bt_mesh_onoff_status onoff_status = { 0 };
-
-	if (!bt_mesh_model_is_extended(srv->ponoff_model)) {
-		srv->onoff.handlers->get(&srv->onoff, NULL, &onoff_status);
-	}
-
-	store(srv, &onoff_status);
+	store_state(srv);
```

Comment:
Not something introduced by this PR, but this storing of the generic onoff value in the ponoff server is getting pretty awkward, and I'm wondering whether it's something we should just stop doing. 

When used in combination with the lightness model, the stored onoff value is not updated when we set the level through the lightness server itself, which really makes this behavior questionable. It gets read when loading the persistent state, though. This produces a glitch where the value gets set twice in quick succession on startup. @PavelVPV changed the settings behavior for scenes in #4327 to not set the scene data for the lower level models, effectively fixing this problem for scenes. I remember having some discussion on doing this for the settings data as well, perhaps it's something we should revisit? As far as I can tell, this storage is currently only interesting when the model isn't extended.